common/app/templates/inlineJS/blocking/analytics.scala.js

```diff
@@ -21,6 +35,46 @@ try {
             return config.page.section || '';
         };
 
+        function forEach(array, callback, scope) {
+            var data = [];
+            for (var i = 0; i < array.length; i++) {
+                data.push(callback.call(scope, i, array[i]));
+            }
+            return data;
+        }
+
+        @* This only works for string arrays *@
+        function uniq(array) {
+            var data = [];
+            for (var i = 0; i < array.length; i++) {
+                if(data.indexOf(array[i]) == -1) {
+                    data.push(array[i]);
+                }
+            }
+            return data;
+        }
+
+        function getUserFromCookie() {
```

Comment:
this functionality is duplicated in `common/app/templates/inlineJS/nonBlocking/getUsername.scala.js` - should extract one out and share